include/illixr/data_loading.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "illixr/csv_iterator.hpp"
+#include "illixr/data_format.hpp"
+#include "illixr/error_util.hpp"
+
+#include <eigen3/Eigen/Dense>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <spdlog/spdlog.h>
+#include <string>
+
+using namespace ILLIXR;
+
+template<typename T>
+static std::map<ullong, T> load_data(const std::string& spath, const std::string& plugin_name,
+                                     std::map<ullong, T> (*func)(std::ifstream&, const std::string&)) {
+    const char* illixr_data_c_str = std::getenv("ILLIXR_DATA");
```

Comment:
For generality, we might consider adding another string argument instead of hard-coding `ILLIXR_DATA`. Maybe same for `data.csv` below.

Reply:
Possibly, but I think that issue would be better addressed in the generalized data set plugin which is under development.